src/openloco/windows/PlayerInfoPanel.cpp

```diff
@@ -0,0 +1,369 @@
+    };
+
+    // 0x00509d08
+    static widget_t _widgets[] = {
+        make_widget({ 0, 0 }, { 140, 29 }, widget_type::wt_3, 0),
+        make_widget({ 2, 2 }, { 136, 25 }, widget_type::wt_3, 0),
+        make_widget({ 1, 1 }, { 26, 26 }, widget_type::wt_9, 0),
+        make_widget({ 27, 2 }, { 111, 12 }, widget_type::wt_9, 0, string_ids::null, string_ids::tooltip_company_value),
+        make_widget({ 27, 14 }, { 111, 12 }, widget_type::wt_9, 0, string_ids::null, string_ids::tooltip_performance_index),
+        widget_end(),
+    };
+
+    static std::map<CorporateRating, string_id> _ratingNames = {
+        { CorporateRating::platelayer, string_ids::str_1772 },
+        { CorporateRating::engineer, string_ids::str_1773 },
+        { CorporateRating::trafficManager, string_ids::str_1774 },
+        { CorporateRating::transportCoordinator, string_ids::str_1775 },
+        { CorporateRating::routeSupervisor, string_ids::str_1776 },
+        { CorporateRating::director, string_ids::str_1777 },
+        { CorporateRating::chiefExecutive, string_ids::str_1778 },
+        { CorporateRating::chairman, string_ids::str_1779 },
+        { CorporateRating::president, string_ids::str_1780 },
+        { CorporateRating::tycoon, string_ids::str_1781 },
+    };
+
+    static window_event_list _events;
+
+    static loco_global<uint16_t, 0x0050A004> _50A004;
+    static loco_global<uint16_t, 0x0052338A> _tooltipTimeout;
+    static loco_global<int32_t, 0x00e3f0b8> gCurrentRotation;
+    static loco_global<uint8_t, 0x00508F1A> game_speed;
+    static loco_global<uint16_t, 0x0113DC78> _113DC78;
+
+    static void prepare_draw(window* window);
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi);
+    static void on_mouse_up(ui::window* window, widget_index widgetIndex);
+    static void on_mouse_down(ui::window* window, widget_index widgetIndex);
+    static void on_dropdown(window* w, widget_index widgetIndex, int16_t item_index);
+    static ui::cursor_id on_cursor(ui::window* window, int16_t widgetIdx, int16_t xPos, int16_t yPos, ui::cursor_id fallback);
+    static void tooltip(FormatArguments& args, ui::window* window, widget_index widgetIndex);
+    static void on_update(window* w);
+    static CorporateRating performanceToRating(int16_t ax);
+
+    // 0x43AA4C
+    static void playerMouseDown(ui::window* self, widget_index widgetIndex)
+    {
+        auto args = FormatArguments();
+        std::vector<const company*> sortedCompanies;
+
+        for (const auto& c : companymgr::companies())
+        {
+            if (!c.empty())
+            {
+                sortedCompanies.push_back(&c);
+            }
+        }
+
+        sort(
+            sortedCompanies.begin(),
+            sortedCompanies.end(),
+            [](const company* a, const company* b) {
+                return a->performance_index > b->performance_index;
+            });
+
+        std::array<string_id, 15> positionArray = {
+            string_ids::position_1st,
```

Comment:
Our macOS CI complains about a lack of brace initialisers. Rather than adding them, I'd prefer to just go with a basic array type instead. The container doesn't serve much of an advantage here, I think.

Reply:
If i had accessed it with .at() it would be bounds checked. Hmm i guess there isn't any benefit.